From 5e97e3b4abb2b9204c598a3495a1cd6f3428c7fb Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Fri, 18 Jan 2013 17:28:39 +0800
Subject: [PATCH 09/95] ENGR00215174 calculate jgeg buffer size correctly.

because now add non HAL_PIXEL_FORMAT_BLOB for jpeg process.
and it is only NV12 format supported now.
calculate jpeg buffer size according to format.

Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 services/camera/libcameraservice/api1/client2/JpegProcessor.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/api1/client2/JpegProcessor.cpp b/services/camera/libcameraservice/api1/client2/JpegProcessor.cpp
index 1849385..825c41d 100644
--- a/services/camera/libcameraservice/api1/client2/JpegProcessor.cpp
+++ b/services/camera/libcameraservice/api1/client2/JpegProcessor.cpp
@@ -247,15 +247,17 @@ status_t JpegProcessor::processNewCapture() {
         return OK;
     }
 #endif
-    // Find size of JPEG image
+    // Find size of JPEG image   
     int size = 0;
     if (imgBuffer.format != HAL_PIXEL_FORMAT_BLOB) {
+        //now only support nv12.
         size = imgBuffer.stride * imgBuffer.height * 3/2;
     }
     else {
         size = mMaxJpegSize;
     }
 
+
         // Find size of JPEG image
     size_t jpegSize = findJpegSize(imgBuffer.data, size/*imgBuffer.width*/);
         if (jpegSize == 0) { // failed to find size, default to whole buffer
-- 
1.8.0

